### PR TITLE
Adding Support for Unity 2020.1 to use Prismatic Joints

### PR DESCRIPTION
**Pull request by chris15joseph**

### CODE REVIEW

**Comment by mrpropellers** [comment]
Hi @chris15joseph, apologies for not addressing this PR sooner.  There are actually a number of critical stability changes to Unity's PhysX integration in addition to this API change that led us to not officially supporting 2020.1. We still have some older ifndefs mentioning earlier versions that need to be cleaned up...

Could you provide more information on why support for 2020.1 is needed for you? That would help us make decisions when we discuss back-porting in the future.

File: Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismatic.cs
Comment (mrpropellers) [inline Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismatic.cs:13]: This should not be required here; it's preferable to allow the package manager to enforce version constraints using the [package.json](https://github.com/Unity-Technologies/URDF-Importer/blob/main/package.json#L6-L7).

**Review by mrpropellers — CHANGES_REQUESTED** [review]

File: Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismatic.cs
Comment (chris15joseph) [reply Runtime/UrdfComponents/UrdfJoints/UrdfJointPrismatic.cs:13]: > This should not be required here; it's preferable to allow the package manager to enforce version constraints using the [package.json](https://github.com/Unity-Technologies/URDF-Importer/blob/main/package.json#L6-L7).

Noted. I will remove these lines.

**Comment by chris15joseph** [comment]
> Could you provide more information on why support for 2020.1 is needed for you? That would help us make decisions when we discuss back-porting in the future.

The client project that I'm working on recently switched from an older version of Unity to version 2020.1. Updating Unity again so soon is not in the plans as this requires a lot of infrastructure changes (at least until there is LTS for version 2020).

**Review by mrpropellers — APPROVED** [review]
Ok, so we discussed this a little more internally... our stance is still that 2020.1 can not be officially supported, however if this change is sufficient to enable some functionality for your use case in the meantime, I'm fine with approving it as long as the version listed in the package.json stays 2020.2. Just bear in mind we will eventually (timeframe tbd) come through and clear out all of the 2020_1 preprocessor directives which will include these changes.

**Review by mrpropellers — CHANGES_REQUESTED** [review]
Sorry, revoked review to request a quick change: This merge should be with dev, not main. We will eventually set up rules to ensure nothing can go straight to main, but for now submissions need to be manually configured to merge to dev.

**Comment by chris15joseph** [comment]
> Ok, so we discussed this a little more internally... our stance is still that 2020.1 can not be officially supported, however if this change is sufficient to enable some functionality for your use case in the meantime, I'm fine with approving it as long as the version listed in the package.json stays 2020.2. Just bear in mind we will eventually (timeframe tbd) come through and clear out all of the 2020_1 preprocessor directives which will include these changes.

Thanks for taking the time to consider this.

**Review by mrpropellers — APPROVED** [review]

**Comment by chris15joseph** [comment]
Feel free to merge this whenever you get the chance. I do not have write access and therefore am unable to merge.

**Comment by mrpropellers** [comment]
> Feel free to merge this whenever you get the chance. I do not have write access and therefore am unable to merge.



> Feel free to merge this whenever you get the chance. I do not have write access and therefore am unable to merge.

Interesting... there must be an organization-level protection on our repos I'm not aware of. I'll go ahead and merge this now!

**Pull request merged**
